pkg/storaged/crypto-keyslots.jsx

```diff
@@ -0,0 +1,556 @@
+                            {_("Confirm removal with passphrase")}
+                        </label>
+                    </div>
+                    <input className="form-control radio-child" type="password" hidden={val === false}
+                           value={val} onChange={event => change(event.target.value)} />
+                    <br />
+                    <div className="radio">
+                        <label>
+                            <input type="radio" checked={val === false}
+                                   onChange={event => change(false)} />
+                            { fmt_to_array(_("Force remove passphrase in $0"), <b>{key_slot}</b>) }
+                        </label>
+                    </div>
+                    <div className="radio-child" hidden={val !== false}>
+                        Removing a passphrase without confirmation can be dangerous.
+                    </div>
+                </div>
+            );
+        }
+    };
+};
+
+function remove_passphrase_dialog(block, key) {
+    dialog_open({ Title: _("Remove passphrase"),
+                  Fields: [
+                      RemovePassphraseField("passphrase", key, block_name(block))
+                  ],
+                  Action: {
+                      DangerButton: true,
+                      Title: _("Remove"),
+                      action: function (vals) {
+                          if (vals.passphrase === false)
+                              return slot_remove(block, key.slot);
+                          else
+                              return passphrase_remove(block, vals.passphrase);
+                      }
+                  }
+    });
+}
+
+const RemoveClevisField = (tag, key, dev) => {
+    return {
+        tag: tag,
+        title: <span className="pficon pficon-warning-triangle-o large" />,
+        options: { },
+        initial_value: "",
+
+        render: (val, change) => {
+            return (
+                <div data-field={tag}>
+                    <div className="large">{ fmt_to_array(_("Remove $0?"), <b>{key.url}</b>) }</div>
+                    <div>{ fmt_to_array(_("Keyserver removal may prevent unlocking $0."), <b>{dev}</b>) }</div>
+                </div>
+            );
+        }
+    };
+};
+
+function remove_clevis_dialog(client, block, key) {
+    dialog_open({ Title: _("Remove Tang keyserver"),
+                  Fields: [
+                      RemoveClevisField("keyserver", key, block_name(block))
+                  ],
+                  Action: {
+                      DangerButton: true,
+                      Title: _("Remove"),
+                      action: function () {
+                          return clevis_remove(block, key);
+                      }
+                  }
+    });
+}
+
+export class CryptoKeyslots extends React.Component {
+    constructor() {
+        super();
+        this.state = { version: 1, slots: null, slot_error: null, max_slots: 8 };
```

Comment:
The `max_slots` is coming from LUKS1's disk format, right? That could use a comment. Is that different with LUKS2?

Reply:
Correct.  I don't know the situation with LUKSv2, but I would expect it to not have a fixed maximum number of slots.